### PR TITLE
Add support for sending JSON record

### PR DESCRIPTION
**Pull request by zyiou**

### CODE REVIEW

File: pkg/exporter/process.go [hunk 1]
```diff
@@ -84,7 +89,9 @@ type ExporterInput struct {
 func InitExportingProcess(input ExporterInput) (*ExportingProcess, error) {
 	var conn net.Conn
 	var err error
-
+	if input.CollectorProtocol == "udp" && input.SendJSONRecord {
```
Comment (srikartati) [inline pkg/exporter/process.go:92]: Hi Yiou, I thought about this more. Even though UDP packet length is large here.. IP fragmentation happens and packet get reassembled in Logstash instance. There is possibility of record loss, but that is ok. Logstash received records fine with IP fragmentation. I checked with large records and UDP transport. 
You can remove this check.

**Review by srikartati — COMMENTED** [review]
LGTM except for comment on UDP transport.
Hope we got a chance to test this on local setup.

File: pkg/exporter/process.go (hunk 1 above)
Comment (zyiou) [reply pkg/exporter/process.go]: Got it. Updated. will add some local tests later.

Comment (srikartati) [reply pkg/exporter/process.go]: I did not mean local unit tests. I just meant a test on local setup with Antrea+local ELK flow collector.

File: pkg/exporter/process.go [hunk 2]
```diff
@@ -31,6 +33,7 @@ import (
 
 const startTemplateID uint16 = 255
 const defaultCheckConnInterval = 10 * time.Second
+const msgBufferLen = 4000
```
Comment (srikartati) [inline pkg/exporter/process.go:36]: we may end up needing a bigger record if metadata in ipfix is long, right?

Comment (zyiou) [reply pkg/exporter/process.go:36]: Correct. Do you have any suggested size in mind?

Comment (srikartati) [reply pkg/exporter/process.go:36]: May be allocate bigger size with a factor based on the length of ipfix message length?

File: pkg/exporter/process.go [hunk 3]
```diff
@@ -305,6 +316,32 @@ func (ep *ExportingProcess) createAndSendMsg(set entities.Set) (int, error) {
 	return bytesSent, nil
 }
 
+// createAndSendJSONMsg takes in a set as input, creates the JSON record, and sends it out.
+func (ep *ExportingProcess) createAndSendJSONMsg(set entities.Set) (int, error) {
+	var bytesSent int
+	for _, record := range set.GetRecords() {
+		records := make(map[string]interface{})
```
Comment (kais66) [inline pkg/exporter/process.go:323]: nit: is "records" an accurate variable name? It seems to be a collection of fields from records.

**Comment by kais66** [comment]
Can we benchmark the record sizes for IPFIX and JSON formats? Some data for a few sample records should suffice. This is to get an understanding how much overhead in terms of additional bytes JSON format is causing.

File: pkg/exporter/process.go (hunk 3 above)
Comment (zyiou) [reply pkg/exporter/process.go]: Updated. Thanks

File: pkg/exporter/process.go [hunk 4]
```diff
@@ -305,6 +327,32 @@ func (ep *ExportingProcess) createAndSendMsg(set entities.Set) (int, error) {
...
+		elements := make(map[string]interface{})
+		for _, element := range record.GetOrderedElementList() {
+			elements[element.Element.Name] = element.Value
+		}
+		message := make(map[string]interface{}, 2)
+		message["ipfix"] = elements
+		message["@timestamp"] = time.Now().Format(time.RFC3339)
+		writer := bytes.NewBuffer(make([]byte, 0, ep.jsonBufferLen))
+		encoder := json.NewEncoder(writer)
+		encoder.Encode(message)
```
Comment (srikartati) [inline pkg/exporter/process.go:343]: What happens if we message buffer in the encoder? Does it return error?

Comment (zyiou) [reply pkg/exporter/process.go:343]: you mean encode(writer)?

Comment (srikartati) [reply pkg/exporter/process.go:343]: I mean encoder.Encode function. Looked into it. I think we need to handle the error here.
https://github.com/square/go-jose/blob/v2.6.0/json/stream.go#L188

**Review by srikartati — APPROVED** [review]
LGTM

**Pull request merged**